docs/world api.md

```diff
@@ -339,6 +314,63 @@ avoiding the need for indirect conditions at the expense of performance.
 An item rule is a function that returns `True` or `False` for a `Location` based on a single item. It can be used to
 reject the placement of an item there.
 
+### Events (or "generation-only items/locations")
+
+An event item or location is one that only exists during multiworld generation. The server is never made aware of them,
+event locations can never be checked by the player, nor can event items be received during play.
+
+Events are used to represent in-game actions (that aren't regular Archipelago locations) when either:
+
+* We want to show in the spoiler log when the player is expected to perform the in-game action.
+* It's the cleanest way to represent how that in-game action impacts logic.
+
+Typical examples include completing the goal, defeating a boss, or flipping a switch that affects multiple areas.
+
+To be precise: the term "event" on its own refers to the special combination of an "event item" placed on an "event
+location". Event items and locations are created the same way as normal items and locations, except that they have an
+`id` of `None` and an event item must be placed on an event location
+(and vice versa). Finally, although events are often described as "fake" items and locations, it's important to
+understand that they are perfectly real during generation.
+
+The most common way to create an event is to create the event item and the event location, then immediately call
+`.place_locked_item()`:
+
+```python
+victory_loc = MyGameLocation(self.player, "Defeat the Final Boss", None, final_boss_arena_region)
+victory_loc.place_locked_item(MyGameItem("Victory", ItemClassification.progression, None, self.player))
+self.multiworld.completion_condition[self.player] = lambda state: state.has("Victory", self.player)
+set_rule(victory_loc, lambda state: state.can_reach_region("Final Boss Arena", self.player))
```

Comment:
```suggestion
set_rule(victory_loc, lambda state: state.has("Boss Defeating Sword", self.player))
```
`state.can_reach_region` on a location's parent region will never return false, and we definitely don't want the docs to have an example like that.